src/utils/helpers/structure/buildSectionsStructureNew.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ApplicationElementStates,
+  ResponsesByCode,
+  SectionDetails,
+  SectionsStructure,
+} from '../../types'
+import { getPageElements } from '../application/getPageElements'
+
+interface BuildSectionsStructureProps {
+  sections: SectionDetails[]
+  elementsState: ApplicationElementStates
+}
+
+/**
+ * @function buildSectionsStructure
+ * Create the sections structure object to be used
+ * in displaying each sections and its pages on either
+ * an application or review pages. The structure is
+ * defined by the SectionStructure type.
+ * @param sections Array with all sections details
+ * @param elementsState object with all elements in application
+ * @returns Object of complete sections structure
+ */
+export const buildSectionsStructure = ({
+  sections,
+  elementsState,
+}: BuildSectionsStructureProps): SectionsStructure => {
+  // Create the sections and pages structure to display each section's element
+  // Will also add the responses for each element, and can add reviews if received by props
+  return sections.reduce((sectionsStructure: SectionsStructure, section) => {
+    const pageNumbers = Array.from(Array(section.totalPages).keys(), (n) => n + 1)
+    const pages = pageNumbers.reduce((pages, pageNumber) => {
+      const elements = getPageElements({
+        elementsState,
+        sectionIndex: section.index,
+        pageNumber,
+      })
+      if (elements.length === 0) return pages
+
+      // Will build the array of elements
+      const state = elements.map((element) => ({ element }))
+
+      const pageName = `Page ${pageNumber}`
+      return { ...pages, [pageName]: { number: pageNumber, state } }
+    }, {})
+
+    return {
+      ...sectionsStructure,
+      [section.code]: {
+        details: section,
+        pages,
+      },
+    }
+  }, {})
+}
+
+export const buildTemplateSectionsStructure = (sections: SectionDetails[]) => {
```

Comment:
This is not being used in the PR, is it used somewhere else ?

Reply:
It is used in the original `buildSectionsStructure` by `ApplicationCreate` container. Not in this one yet, but should be when we remove the old file. Should I move to a separated file? Oh, I'm also going to rename this to `buildSectionsStructureNEW` now.